### ISSUE TITLE
Glue Crawler Role reference E3008 error since 0.23.4

### ISSUE THREAD

**Issue by davidpricedev** [aws-cloudformation/cfn-python-lint#1119]
*cfn-lint version: (`cfn-lint --version`)*
0.23.4 and 0.23.5

*Description of issue.*

These are some snippets from my CFN template: 
Role:
```yaml
  GlueCrawlerRole:
    Type: AWS::IAM::Role
```
Crawler (also tried `!GetAtt GlueCrawlerRole.Arn` to produce the other error):
```yaml
  MyCrawler:
    Type: AWS::Glue::Crawler
    Properties:
      Role: !Ref GlueCrawlerRole
```

running cfn-lint on that template generates the following errors on the `Role:` line of `MyCrawler` when using !Ref and !GetAtt respectively. 

> E3008 Property "Role" can Ref to resources of types [AWS::IAM::InstanceProfile] at Resources/MyCrawler/Properties/Role/Ref ./my_cfn.yaml:405:7
 
> E3008 Property "Role" can Fn::GetAtt to a resource of types [AWS::IAM::InstanceProfile] at Resources/MyCrawler/Properties/Role/Fn::GetAtt my_cfn.yaml:405:7

I can successfully deploy the template with either the !Ref or !GetAtt form to AWS. So I believe the error message is incorrect/false-positive. When I roll back to 0.23.3 (and presumably earlier) the linter passes my cfn template successfully.

**Comment by kddejong** [comment]
Think we should have this fixed in https://github.com/aws-cloudformation/cfn-python-lint/pull/1114 waiting on release.

**Comment by kddejong** [comment]
Fixed in v0.24.0 and released.
